### ISSUE TITLE
sincos leads to stackoverflow

### ISSUE THREAD

**Issue by AwesomeQuest** [dzhang314/MultiFloats.jl#44]
Because of this bit of code
```julia
_sincos(x::AbstractFloat) = sincos(x)
_sincos(x) = (sin(x), cos(x))

sincos(x) = _sincos(float(x))
```
in
`julia\base\special\trig.jl`

Calling sincos with a MultiFloat causes a stackoverflow.

Minimum working example
```julia
using MultiFloats
sincos(Float64x2(4))
ERROR: StackOverflowError:
Stacktrace:
     [1] _sincos(x::MultiFloat{Float64, 2})
       @ Base.Math .\special\trig.jl:203
     [2] sincos(x::MultiFloat{Float64, 2})
       @ Base.Math .\special\trig.jl:206
--- the last 2 lines are repeated 39990 more times ---
 [79983] _sincos(x::MultiFloat{Float64, 2})
       @ Base.Math .\special\trig.jl:203
```

I'm trying to use `conv` from DSP.jl and GenericFFT.jl, is there a workaround for this? I don't have a firm grasp on how to change the method used by conv in a scenario like this.

P.S. Why would one write code like the above, it seems designed to make a stack overflow in the case that calling `float` does nothing

**Comment by dzhang314** [comment]
Hi @AwesomeQuest, thanks for your interest in **MultiFloats.jl**! This is very strange code in `Base`, and I don't understand why `Base.sincos` is implemented this way, either. For a workaround, you can simply overload `Base.sincos` yourself, by adding the following code after you import `MultiFloats`:

```
Base.sincos(x::MultiFloat{T,N}) where {T,N} = MultiFloat{T,N}.(
    MultiFloats._call_big(sincos, x, precision(MultiFloat{T,N}) + 20))
```

This will call the `BigFloat` implementation of `sincos`. Here, the `+ 20` specifies a number of extra bits of precision -- you can set it lower for a (marginal) speed improvement, but I don't recommend setting it higher, since any additional bits are likely to be cut off when re-converting back to `Float64xN`.

**Comment by dzhang314** [comment]
Whoops, accidentally closed in my last comment.

I'll also add overloads for `sincos`, `sincosd`, and `sincospi` in the next patch release for **MultiFloats.jl**, which will automatically be defined when you call `use_bigfloat_transcendentals`, so this manual workaround won't be necessary in the future. Thanks for pointing this out!

**Comment by dzhang314** [comment]
Fixed in [commit 1d54726](https://github.com/dzhang314/MultiFloats.jl/commit/1d547265c8e2db1a1904aea782bf201fb440affb).
